st2common/st2common/validators/api/reactor.py

```diff
@@ -36,3 +45,29 @@ def validate_criteria(criteria):
         if pattern is None:
             raise ValueValidationException('For field: ' + key + ', no pattern specified ' +
                                            'for operator ' + operator)
+
+
+def validate_trigger_parameters(trigger_db):
+    """
+    This function validates parameters for system triggers (e.g. timers).
+
+    Note: Eventually we should also validate parameters for user defined triggers which correctly
+    specify JSON schema for the parameters.
+
+    :param trigger_db: Trigger DB object.
+    :type trigger_db: :class:`TriggerDB`
+    """
+    if not trigger_db:
+        return None
+
+    trigger_type_ref = trigger_db.type
+    parameters = trigger_db.parameters
+
+    if trigger_type_ref not in SYSTEM_TRIGGER_TYPES:
```

Comment:
Yep - see my note in PR.

We would need to go over trigger definitions in st2contrib since I believe a lot of them are out of date or wrong.